### PR TITLE
chore(DatasourceEditor): Create Datasource Legacy Editor Feature Flag

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

This creates a new feature flag for the legacy datasource editor that we are going to depreciate in superset 2.0. Currently it is set to False, which will show the legacy datasource editor, but it gives users the option to not have the option available anymore. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by geido — APPROVED** [review]
LGTM! CI might require some love though

**Comment by AAfghahi** [comment]
> LGTM! CI might require some love though

Thank you! Just fixed CI issues (crossing fingers)

File: superset-frontend/src/components/Datasource/DatasourceModal.test.jsx [hunk 1]
```diff
@@ -72,7 +72,9 @@ describe('DatasourceModal', () => {
   beforeEach(async () => {
     isFeatureEnabledMock = jest
       .spyOn(featureFlags, 'isFeatureEnabled')
-      .mockReturnValue(true);
+      .mockImplementation(
+        featureFlag => featureFlag === FeatureFlag.ENABLE_REACT_CRUD_VIEWS,
```
Comment (hughhhh) [inline superset-frontend/src/components/Datasource/DatasourceModal.test.jsx:76]: don't we want `DISABLE_LEGACY_DATASOURCE_EDITOR` here as well?

**Comment by hughhhh** [comment]
Approved 🚀

**Pull request merged**